app_dart/lib/src/service/scheduler.dart

```diff
@@ -154,11 +155,24 @@ class Scheduler {
       log.severe('Failed to add commit ${commit.sha!}: $error');
     }
 
-    await luciBuildService.schedulePostsubmitBuilds(commit: commit, toBeScheduled: toBeScheduled);
-
+    await _batchScheduleBuilds(commit, toBeScheduled);
     await _uploadToBigQuery(commit);
   }
 
+  /// Schedule all builds in batch requests instead of a single request.
+  ///
+  /// Each batch request contains [Config.batchSize] builds to be scheduled.
+  Future<void> _batchScheduleBuilds(Commit commit, List<Tuple<Target, Task, int>> toBeScheduled) async {
+    final List<Future> futures = <Future>[];
+    for (int i = 0; i < toBeScheduled.length; i += config.batchSize) {
+      futures.add(luciBuildService.schedulePostsubmitBuilds(
+        commit: commit,
+        toBeScheduled: toBeScheduled.sublist(i, min(i + config.batchSize, toBeScheduled.length)),
```

Comment:
is sublist upper bound index non-inclusive?

Reply:
That's correct.